app/controllers/v0/sessions_controller.rb

```diff
@@ -73,10 +78,14 @@ def parse_current_loa
       LOA::MAPPING[raw_loa]
     end
 
-    def create_new_user
-      user = User.new(user_attributes)
-      user = Decorators::MviUserDecorator.new(user).create unless user.loa1?
-      user
+    def saml_user
+      @user || create_saml_user
```

Comment:
I think you also want:

`@saml_user ||= create_saml_user` Instead of:
`@saml_user || create_saml_user` if you're trying to memoize it.